### DISCUSSION TITLE
[TransferList] Different listComponent for both RenderList

### DISCUSSION THREAD

**Discussion by agong358** [mantinedev/mantine#4689] [Feature requests]
We're currently working on integrating the `TransferList` component with the `InfiniteScroll` component from the [infinite-scroll-component](https://github.com/ankeetmaini/react-infinite-scroll-component) library. The reason why we want to use an `InfiniteScroll` is because we're displaying a lot of data to populate the list from the back-end, and want to use lazy loading. 

Right now, we're passing the `InfiniteScroll` as a `listComponent` to `TransferList`. However, since the `listComponent` is a shared prop for both `RenderList`s, the `InfiniteScroll` is applied on both components. This is undesirable because only the left `RenderList` should be able to trigger the queries to fetch more items. However, since the `InfiniteScroll` is also in the right `RenderList`, it can also trigger these calls. In this scenario, we would want the `listComponent` of the left list to be `InfiniteScroll`. As for the right list, we would simply use the default component.

For that reason, would it be possible to allow two different `listComponent` for both `RenderLists`? By default, we could have them share the same `listComponent`. But say we have `leftListComponent` and `rightListComponent`, we would use those props instead.

I would also be willing to implement it myself if there is no objection for this!

**Comment by rtivital** [comment]
It is available in v7 – https://v7.mantine.dev/combobox?e=TransferList
You can completely custom transfer lists with Combobox – https://v7.mantine.dev/core/combobox

**Reply by agong358** [reply]
I'll check it out, thanks for the quick response!

**Comment by agong358** [comment]
I see that version 7 is still in beta. Is there a similar solution or a workaround in the current latest stable version?

**Reply by rtivital** [reply]
No, nothing similar is available in v6
